sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryIO.java

```diff
@@ -1588,20 +1615,19 @@ public static Bound to(TableReference table) {
      * <p>{@code tableSpecFunction} should be deterministic. When given the same window, it should
      * always return the same table specification.
      */
-    public static Bound to(SerializableFunction<BoundedWindow, String> tableSpecFunction) {
-      return new Bound().to(tableSpecFunction);
+    public static Bound to(TableWritePolicy<String> tableRefPolicy) {
```

Comment:
tableWritePolicy perhaps?

In general I'd suggest to rename the concept to something else - "table write policy" is overly generic. Maybe something with the word "routing" or "destination"? e.g. call it DestinationTableFn?